src/zoom_type.h

```diff
@@ -15,6 +15,8 @@
 static uint const ZOOM_LVL_SHIFT = 2;
 static uint const ZOOM_LVL_BASE  = 1 << ZOOM_LVL_SHIFT;
 
+static const int8 ZOOM_LVL_GUI_AUTO = -1;
```

Comment:
Putting this out of context of `ZOOM_LVL_MIN` is a bit awkward, as core earlier assumes that `ZOOM_LVL_GUI_AUTO` is below `ZOOM_LVL_MIN` (as any value below `ZOOM_LVL_MIN` assumes it was auto-detect). But here there is no such link or indication the values are linked like this. Can I suggest to either use `ZOOM_LVL_MIN` in the value of `ZOOM_LVL_GUI_AUTO`, or compare earlier against this value (instead of `< ZOOM_LVL_MIN`?)